crates/rpc-types-eth/src/transaction/error.rs

```diff
@@ -58,6 +58,15 @@ pub enum ConversionError {
     /// Missing block number
     #[error("missing block number")]
     MissingBlockNumber,
+    /// Missing source hash (for Optimism deposit tx)
+    #[error("missing source hash")]
+    MissingSourceHash,
+    /// Invalid source hash (for Optimism deposit tx)
+    #[error("invalid source hash")]
+    InvalidSourceHash,
+    /// Invalid mint value (for Optimism deposit tx)
+    #[error("invalid mint value")]
+    InvalidMintValue,
```

Comment:
these are OP specific ones, I don't like this because this OP specific and should not be included here

we should either add a custom(String) or Box<dyn>

Reply:
would love you feedback about the discussion in the comments